From reinette.chatre@intel.com  Tue Dec  1 16:35:13 2009
From: Reinette Chatre <reinette.chatre@intel.com>
Date: Wed, 25 Nov 2009 11:03:48 -0800
Subject: iwlwifi: Fix issue on file transfer stalled in HT mode
To: stable@kernel.org
Cc: Reinette Chatre <reinette.chatre@intel.com>, wey-yi.w.guy@intel.com, linville@tuxdriver.com
Message-ID: <1259175828-14489-3-git-send-email-reinette.chatre@intel.com>


From: Wey-Yi Guy <wey-yi.w.guy@intel.com>

commit d01032e4fd33110f9f3a085a36cb819c1dfc5827 upstream

Turn on RTS/CTS for HT to prevent uCode TX fifo underrun

This is fix for
http://bugzilla.intellinuxwireless.org/show_bug.cgi?id=2103

Signed-off-by: Wey-Yi Guy <wey-yi.w.guy@intel.com>
Tested-by: Jiajia Zheng <jiajia.zheng@intel.com>
Signed-off-by: Reinette Chatre <reinette.chatre@intel.com>
Signed-off-by: John W. Linville <linville@tuxdriver.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/net/wireless/iwlwifi/iwl-1000.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/net/wireless/iwlwifi/iwl-1000.c
+++ b/drivers/net/wireless/iwlwifi/iwl-1000.c
@@ -71,5 +71,6 @@ struct iwl_cfg iwl1000_bgn_cfg = {
 	.need_pll_cfg = true,
 	.max_ll_items = OTP_MAX_LL_ITEMS_1000,
 	.shadow_ram_support = false,
+	.use_rts_for_ht = true, /* use rts/cts protection */
 };
 
